### PR TITLE
CommandBar: Fix high contrast styling for disabled CommandBar

### PR DESCRIPTION
**Pull request by tomi-msft**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes https://dev.azure.com/microsoftdesign/fluent-ui/_workitems/edit/7813
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Disabled CommandBar was not adopting high contrast styling. This fix adds styling to allow CommandBar to adopt high contrast styling.

**Before**
image

**After**
image

**Also**
Because I have modified SplitButton.styles, I will fix some HC styling in the PR as well for SplitButton.

**Before**
image

**After**
image

Main files to check are BaseButton.styles.ts, CommandBarButton.styles.ts, SplitButton.styles.ts. Rest are snapshot updates.
#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by xugao** [comment]
Tomi, thanks for fixing the CommandBar! it looks good. However i noticed SplitButton is quite broken: http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/14215/merge/fabric-website/dist/index.html?devhost#/controls/web/button

It's already quite broken without your changes. However, you made some styles update to SplitButton, which made the disabled state slightly worse. i think we should just fix SplitButton (in disabled state at least) along with this change

**Comment by tomi-msft** [comment]
Will do!

File: packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts [hunk 1]
```diff
@@ -40,14 +47,17 @@ export const getStyles = memoizeFunction(
...
+                  border: '1px solid WindowText',
+                  borderRightWidth: '0',
                   MsHighContrastAdjust: 'none',
                 },
+                ...getEdgeChromiumNoHighContrastAdjustSelector(),
               },
             },
             '.ms-Button--primary + .ms-Button': {
-              border: 'none',
+              border: '1px solid WindowText',
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts:60]: should not be changing non-HC styles

File: packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts [hunk 2]
```diff
@@ -40,14 +47,17 @@ export const getStyles = memoizeFunction(
...
                 [HighContrastSelector]: {
-                  color: 'Window',
-                  backgroundColor: 'WindowText',
+                  color: 'WindowText',
+                  backgroundColor: 'Window',
+                  border: '1px solid WindowText',
+                  borderRightWidth: '0',
                   MsHighContrastAdjust: 'none',
                 },
+                ...getEdgeChromiumNoHighContrastAdjustSelector(),
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts:56]: i dont think this is needed, can you remove and check?

File: packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts [hunk 3]
```diff
@@ -183,6 +201,7 @@ export const getStyles = memoizeFunction(
             borderColor: 'GrayText',
             backgroundColor: 'Window',
           },
+          ...getEdgeChromiumNoHighContrastAdjustSelector(),
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts:204]: pls check if this is needed too

File: packages/office-ui-fabric-react/src/components/Button/CommandBarButton/CommandBarButton.styles.ts [hunk 4]
```diff
@@ -209,6 +225,9 @@ export const getStyles = memoizeFunction(
 
       menuIcon: {
         color: p.neutralSecondary,
+        [HighContrastSelector]: {
+          color: 'GrayText',
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Button/CommandBarButton/CommandBarButton.styles.ts:229]: Looks like it's missing to set `border: none` here:
image

File: packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts [hunk 5]
```diff
@@ -158,11 +168,19 @@ export const getStyles = memoizeFunction(
...
+              },
+            },
+          },
           [HighContrastSelector]: {
-            border: `1px solid GrayText`,
             color: 'GrayText',
+            border: '1px solid GrayText',
             backgroundColor: 'Window',
           },
+          ...getEdgeChromiumNoHighContrastAdjustSelector(),
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts:183]: pls also check if this is needed

File: packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts (hunk 3 above)
Comment (tomi-msft) [reply packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts:204]: This is needed to fix the styling for the disabled splitbutton example. It is also applied to the span divider element in the disabled commandbar example, so I will need to add a style override in commandbar.styles that removes this to achieve the expected behavior

File: packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts (hunk 2 above)
Comment (tomi-msft) [reply packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts:56]: not needed, removed

File: packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts (hunk 5 above)
Comment (tomi-msft) [reply packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts:179]: not needed, removed

File: packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts [hunk 6]
```diff
@@ -40,15 +47,14 @@ export const getStyles = memoizeFunction(
...
-                  backgroundColor: 'WindowText',
+                  color: 'WindowText',
+                  backgroundColor: 'Window',
+                  border: '1px solid WindowText',
+                  borderRightWidth: '0',
                   MsHighContrastAdjust: 'none',
                 },
               },
             },
-            '.ms-Button--primary + .ms-Button': {
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts:49]: you cannot remove this block, it's breaking non-HC: https://screener.io/v2/states/Skgr32fUL.microsoft-fluentui/a11y-commandbar

Comment (tomi-msft) [reply packages/office-ui-fabric-react/src/components/Button/SplitButton/SplitButton.styles.ts:49]: whoops!

File: packages/office-ui-fabric-react/src/components/Button/CommandBarButton/CommandBarButton.styles.ts (hunk 4 above)
Comment (tomi-msft) [reply packages/office-ui-fabric-react/src/components/Button/CommandBarButton/CommandBarButton.styles.ts:245]: This does not affect the border styling for the icon, SplitButtonMenuButtonDisabled does, updated to remove border

File: apps/a11y-tests/src/tests/__snapshots__/ComponentExamples.test.tsx.snap [hunk 7]
```diff
@@ -6,25 +6,13 @@ exports[`a11y test checks accessibility of ActivityItem (ActivityItem.Compact.Ex
 
 exports[`a11y test checks accessibility of ActivityItem (ActivityItem.Persona.Example) 1`] = `Array []`;
```
Comment (xugao) [inline apps/a11y-tests/src/tests/__snapshots__/ComponentExamples.test.tsx.snap:7]: revert this file

**Pull request merged**